src/SFA.DAS.Commitments.Application/Commands/UpdateApprenticeshipStopDate/UpdateApprenticeshipStopDateCommandHandler.cs

```diff
@@ -0,0 +1,109 @@
+
+namespace SFA.DAS.Commitments.Application.Commands.UpdateApprenticeshipStopDate
+{
+    public sealed class UpdateApprenticeshipStopDateCommandHandler : AsyncRequestHandler<UpdateApprenticeshipStopDateCommand>
+    {
+        private readonly ICommitmentRepository _commitmentRepository;
+        private readonly IApprenticeshipRepository _apprenticeshipRepository;
+        private readonly UpdateApprenticeshipStopDateCommandValidator _validator;
+        private readonly ICurrentDateTime _currentDate;
+        private readonly ICommitmentsLogger _logger;
+        private readonly IHistoryRepository _historyRepository;
+
+        private readonly IAcademicYearValidator _academicYearValidator;
+
+        public UpdateApprenticeshipStopDateCommandHandler(
+            ICommitmentRepository commitmentRepository,
+            IApprenticeshipRepository apprenticeshipRepository,
+            UpdateApprenticeshipStopDateCommandValidator validator,
+            ICurrentDateTime currentDate,
+            ICommitmentsLogger logger,
+            IHistoryRepository historyRepository,
+            IAcademicYearValidator academicYearValidator
+            )
+        {
+            _commitmentRepository = commitmentRepository;
+            _apprenticeshipRepository = apprenticeshipRepository;
+            _validator = validator;
+            _currentDate = currentDate;
+            _logger = logger;
+            _historyRepository = historyRepository;
+            _academicYearValidator = academicYearValidator;
+        }
+
+        protected override async Task HandleCore(UpdateApprenticeshipStopDateCommand command)
+        {
+            _logger.Info($"Employer: {command.AccountId} has called StopApprenticeshipCommand", command.AccountId, apprenticeshipId: command.ApprenticeshipId, caller: command.Caller);
+
+            var validationResult = _validator.Validate(command);
+
+            if (!validationResult.IsValid)
+                throw new ValidationException(validationResult.Errors);
+
+            var apprenticeship = await _apprenticeshipRepository.GetApprenticeship(command.ApprenticeshipId);
+
+            var commitment = await _commitmentRepository.GetCommitmentById(apprenticeship.CommitmentId);
+
+            CheckAuthorization(command, commitment);
+
+            ValidateChangeDateForStop(command.StopDate, apprenticeship);
+
+            await SaveChange(command, commitment, apprenticeship);
+        }
+
+        private async Task SaveChange(UpdateApprenticeshipStopDateCommand command, Commitment commitment, Apprenticeship apprenticeship)
+        {
+            var historyService = new HistoryService(_historyRepository);
+
+            historyService.TrackUpdate(apprenticeship, ApprenticeshipChangeType.ChangeOfStopDate.ToString(), null, apprenticeship.Id, CallerType.Employer, command.UserId, apprenticeship.ProviderId, apprenticeship.EmployerAccountId, command.UserName);
+            apprenticeship.PaymentStatus = PaymentStatus.Withdrawn;
```

Comment:
This doesn't look like it will do anything, other than change the value of this object for the very short remainder of its life. In any case, I don't think it's needed, as the apprenticeship should already be stopped.

Reply:
defo - removed.